core/registration.c

```diff
@@ -278,11 +286,26 @@ static uint8_t prv_register(lwm2m_context_t * contextP,
         lwm2m_free(query);
         return COAP_503_SERVICE_UNAVAILABLE;
     }
-
+    if (contextP->remaining_dataLen > MAX_BLOCK1_SIZE)
```

Comment:
I'd like to see the size determination for whether or not to use Block1 separate from the Block1 size. For example, it might be possible to send a 1400 byte payload in a single message without using Block1. Block1 support is also optional, so it could prevent registering with a server that would succeed with a slightly larger message.

Reply:
I am sorry, but this is not clear to me. How do we decide when to use block transfer ? Do we initiate a transfer of 1400 bytes in one go and if it fails, fallback to block transfer ?

Reply:
I was just thinking a separate #define to determine whether or not to use Block1. If the total payload size is greater, then use Block1. Otherwise send it in one go. Make sure both that and MAX_BLOCK1_SIZE can be overridden by a define on the command line. The Internet generally has an MTU about 1500 bytes, but something like 6LoWPAN would be much smaller.